client/src/helpers/utility.ts

```diff
@@ -72,23 +82,66 @@ const getElementCorners = (
 
 export const getClosestCorner = (
   e: Element | null,
+  elements: { [id: string]: Element },
   xPos: number,
   yPos: number
 ): Corner | null => {
   if (!e) return null;
   if (e.type === ElementType.Polyline) {
     // TODO: Ideally polyline and rect should not share this func
     const matchRadius = 10;
-    const inLeftPoint =
-      Math.abs(e.points[0] + e.points[1] - (xPos + yPos)) < matchRadius;
-    const inRightPoint =
-      Math.abs(e.points[2] + e.points[3] - (xPos + yPos)) < matchRadius;
+    // TODO: Take into account the polyline connectedElement.
+    let inLeftPoint = false;
+    let inRightPoint = false;
+    e.points.forEach((p, i) => {
+      let inPoint;
+      if (p.connectingElementId) {
```

Comment:
also add a tests case for when the element its connected to gets removed and we revert to displaying the origin x,y positions.